activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -230,7 +230,15 @@ def column_for_attribute(name)
     #   person.respond_to(:nothing) # => false
     def respond_to?(name, include_private = false)
       return false unless super
-      name = name.to_s
+
+      case name
```

Comment:
The only time we explicitly call `reponds_to` in the code uses those two symbols. Yes it knows a lot about the ecosystem, it's an optimization. This code gets called many times with those two symbols, and the knowledge of this saves us a good amount of memory and time. With this change, you're looking at 71,136 bytes of memory and 1,777 objects saved per request:

```ruby
require 'benchmark/ips'

Benchmark.ips do |x|
  x.report("Symbol#to_s") {
    name = :to_partial_path.to_s
  }
  x.report("case") {
    case :to_partial_path
    when :to_partial_path
      name = "to_partial_path".freeze
    end
  }
end
```

Also a pretty nice speed boost

```
Calculating -------------------------------------
         Symbol#to_s   122.411k i/100ms
                case   132.355k i/100ms
-------------------------------------------------
         Symbol#to_s      5.456M (±11.1%) i/s -     26.930M
                case      7.838M (±12.5%) i/s -     38.515M
```

Or roughly 43% faster.

I encourage you to doubt people's "performance optimizations", to do so you must argue that the benefit is not worth the cost (slightly more code). Please ping me on a PR If you re-write this to be faster without sacrificing purity or readability,